lib/engine/game/g_1824/game.rb

```diff
@@ -498,7 +498,11 @@ def kk
         end
 
         def exchangable_for_mountain_railway?(player, corporation)
-          corporation.type == :major && @companies.find { |c| mountain_railway?(c) && c.owned_by?(player) }
+          shares_exchangable?(corporation) && @companies.find { |c| mountain_railway?(c) && c.owned_by?(player) }
```

Comment:
You're not using the return value from `@companies.find` except to see if anything was found. `any?` is the correct way to test this.
```suggestion
          shares_exchangable?(corporation) && @companies.any? { |c| mountain_railway?(c) && c.owned_by?(player) }
```